### PR TITLE
rework the api gateway docs

### PR DESCRIPTION
**Pull request by HarshCasper**

### CODE REVIEW

File: content/en/user-guide/aws/apigateway/index.md [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+  --memory-size 128 \
+  --zip-file fileb://function.zip \
+  --role arn:aws:iam::111111111111:role/apigw
+{{< /command >}}
+
+This creates a new Lambda function named `apigw-lambda` with the code you specified.
+
+### Create a REST API
+
+We will use the API Gateway's [`CreateRESTAPI`](https://docs.aws.amazon.com/apigateway/latest/api/API_CreateRestApi.html) API to create a new REST API. Here's an example command:
```
Comment (whummer) [inline content/en/user-guide/aws/apigateway/index.md:71]:
```suggestion
We will use the API Gateway's [`CreateRestApi`](https://docs.aws.amazon.com/apigateway/latest/api/API_CreateRestApi.html) API to create a new REST API. Here's an example command: 
```

File: content/en/user-guide/aws/apigateway/index.md [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+            "id": "u53af9hm83",
+            "path": "/"
+        }
+    ]
+}
+```
+
+Note the ID of the root resource returned in the response. You'll need this ID for the next step.
+
+### Create a Resource
```
Comment (whummer) [inline content/en/user-guide/aws/apigateway/index.md:119]:
```suggestion
### Create a resource
```

File: content/en/user-guide/aws/apigateway/index.md [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+{{< /command >}}
+
+The above command integrates the `GET` method with the Lambda function created in the first step. We can now proceed with the deployment before invoking the API.
+
+### Create a deployment
+
+Create a new deployment for the API using the [`CreateDeployment`](https://docs.aws.amazon.com/apigateway/latest/api/API_CreateDeployment.html) API:
+
+{{< command >}}
+awslocal apigateway create-deployment \
```
Comment (whummer) [inline content/en/user-guide/aws/apigateway/index.md:189]: Would be good to be consistent in the use of the `$`
```suggestion
$ awslocal apigateway create-deployment \
```

File: content/en/user-guide/aws/apigateway/index.md [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+  --rest-api-id <REST_API_ID> \
+  --stage-name test
+{{< /command >}}
+
+Your API is now ready to be invoked. You can use `cURL` or any HTTP REST client to invoke the API endpoint:
+
+{{< command >}}
+curl -X GET http://localhost:4566/restapis/<REST_API_ID>/test/_user_request_/test
```
Comment (whummer) [inline content/en/user-guide/aws/apigateway/index.md:197]:
```suggestion
$ curl -X GET http://localhost:4566/restapis/<REST_API_ID>/test/_user_request_/test
```

File: content/en/user-guide/aws/apigateway/index.md [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+
+The above command integrates the `GET` method with the Lambda function created in the first step. We can now proceed with the deployment before invoking the API.
+
+### Create a deployment
+
+Create a new deployment for the API using the [`CreateDeployment`](https://docs.aws.amazon.com/apigateway/latest/api/API_CreateDeployment.html) API:
+
+{{< command >}}
+awslocal apigateway create-deployment \
+  --rest-api-id <REST_API_ID> \
```
Comment (whummer) [inline content/en/user-guide/aws/apigateway/index.md:190]: Should we try to make the commands fully copy/pasteable top to bottom? I.e., we could assign a variable `$restApiId` from the output above, and then use it here as a reference..

File: content/en/user-guide/aws/apigateway/index.md [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+}
+{{< / command >}}
+
+{{< alert title="Note">}}
+Setting the API Gateway ID via `_custom_id_` works only on the creation of the resource, but not on update in LocalStack. Ensure that you set the `_custom_id_` tag on creation of the resource.
+{{< /alert >}}
+
+## Custom Domain Names with API Gateway
+
+You can use custom domain names with API Gateway V1 and V2 APIs. To route requests to a custom domain name for an API Gateway V2 API, include the Host header with the custom domain name in your request. For example, assuming that you have set up a custom domain name test.`example.com` to point to your LocalStack instance, you can send a request like this:
```
Comment (whummer) [inline content/en/user-guide/aws/apigateway/index.md:355]:
```suggestion
You can use custom domain names with API Gateway V1 and V2 APIs. To route requests to a custom domain name for an API Gateway V2 API, include the `Host` header with the custom domain name in your request. For example, assuming that you have set up a custom domain name `test.example.com` to point to your LocalStack instance, you can send a request like this:
```

File: content/en/user-guide/aws/apigateway/index.md [hunk 7]
```diff
@@ -0,0 +1,389 @@
...
+
+{{< command >}}
+curl -H 'Host: test.example.com' http://localhost:4566/test
```
Comment (whummer) [inline content/en/user-guide/aws/apigateway/index.md:358]:
```suggestion
$ curl -H 'Host: test.example.com' http://localhost:4566/test
```

File: content/en/user-guide/aws/apigateway/index.md [hunk 8]
```diff
@@ -0,0 +1,389 @@
...
+- **Navigate the Stages**: Click on **Stages** tab to view the stages associated with the API, along with their details, such as `Deployment Id`, `Stage Name`, `Client Certificate Id`, and more.
+- **Delete API**: Delete the API configuration (`V1`/`V2`) by selecting the resource, clicking on **Remove Selected** button on top-right and confirming the deletion by clicking on **Continue** button.
+
+You can also use the Resource Browser to check out the **Authorizers**, **Models**, **Request Validators**, **API Keys**, and **Usage Plans**.
+
+## Examples
+
+The following code snippets and sample applications provide practical examples of how to use ACM in LocalStack for various use cases:
+
+- [API Gateway with Custom Domains over our LocalStack Pro samples](https://github.com/localstack/localstack-pro-samples/tree/master/apigw-custom-domain).
```
Comment (whummer) [inline content/en/user-guide/aws/apigateway/index.md:383]:
```suggestion
- [API Gateway with Custom Domains over our LocalStack Pro samples](https://github.com/localstack/localstack-pro-samples/tree/master/apigw-custom-domain)
```

File: content/en/user-guide/aws/apigateway/index.md [hunk 9]
```diff
@@ -0,0 +1,389 @@
...
+- [Websockets via API Gateway V2](https://github.com/localstack/localstack-pro-samples/tree/master/serverless-websockets).
```
Comment (whummer) [inline content/en/user-guide/aws/apigateway/index.md:384]:
```suggestion
- [Websockets via API Gateway V2](https://github.com/localstack/localstack-pro-samples/tree/master/serverless-websockets)
```

**Review by whummer — APPROVED** [review]
Nice work @HarshCasper , very detailed, thoroughly structured docs. ✨ Only added a few minor comments/suggestions..

File: content/en/user-guide/aws/apigateway/index.md (hunk 5 above)
Comment (HarshCasper) [reply content/en/user-guide/aws/apigateway/index.md:190]: Will take this up on a second iteration 👍

**Pull request merged**